oldjobclient.go

```diff
@@ -0,0 +1,206 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+// OldJobClient fetches retired jobs from persistent storage (e.g. S3) that
+// stores job indefinitely (or a very long time)
+type OldJobClient interface {
+	FetchJob(id string) (*job, error)
+	FetchFlowJobIds(flowID string) ([]string, error)
+}
+
+/**
+ * We expect the jobs to be stored in the bucket with the following structure:
+ *
+ * <s3bucket>/
+ * 		<jobsPrefix>/
+ *    	<jobid>.json
+ * 			<otherjobid>.json
+ * 		<flowPrefix>/
+ * 			<flowid>/
+ *				<jobid>.json
+ *			<otherflowid>/
+ *				<otherjobid>.json
+ */
+type s3JobClient struct {
```

Comment:
What do you think of hiding the S3 implementation detail where practical? This lets us e.g. swap out our persistence layer or add support for additional clients without a crufted API or large refactor.